### ISSUE TITLE
Maximum denormalization level, how to fail on missing entity?

### ISSUE THREAD

**Issue by brabeji** [gpbl/denormalizr#5]
My API gives me one level of nested associations in general though I have complete structure of `normalizr` schemas assembled so denormalization often fails with `Cannot read property 'x' of undefined`.

Options discussed elsewhere (if i recall correctly):

1. Implement sort of `limit` param of `denormalize` to restrict nesting
2. Externally trim schemas to basically implement 1.
3. Fail silently

For now I will try second approach and see how it works out. 
Any advices much appreciated!

**Comment by gpbl** [comment]
I'd like to understand better your use case: it could happen when a schema has an _optional_ entity in it. Is this your case? I wonder if ignoring undefined entities would not swallow errors that should be actually thrown.
What about creating a second schema without the optional entity?

**Comment by brabeji** [comment]
Indeed, it is exactly my use case.

> What about creating a second schema without the optional entity?

That is in other words what I meant by 2. so I guess it's ok.

Should we close this or consider adding some helper function like `getTrimmedSchema(schema, maxDepth)`? Not sure if it fits to scope of `denormalizr`.

**Comment by gpbl** [comment]
Wait I see other benefits for handling missing entities :)

Here is another similar case I've just met, which is more common in my code: I'm loading a route displaying `MyItemComponent` which should display an item fetched later using route's params:

```js
// ... MyItemComponent class... 

const mapStateToProps = (state, ownProps) => ({
  shout: state.entities.myItem[ownProps.params.itemId] ?
    denormalize(state.entities.myItems[ownProps.params.id], state.entities, myItemsSchema) : 
    undefined,
});

export default connect(mapStateToProps)(MyItemComponent);
```

if `denormalize` could handle undefined entities, I could just write:

```js
const mapStateToProps = (state, ownProps) => ({
  shout: denormalize(state.entities.myItems[ownProps.params.id], state.entities, myItemsSchema)
});
```

I think it would be a good idea. Being a `0.0.x` release we should feel free to experiment :P

**Comment by gpbl** [comment]
Anyway, I don't think this is much related to the number of denormalized nested entities.

**Comment by brabeji** [comment]
I am doing the same and it seems fine for now

[My first crude stab at `getTrimmedSchema`](https://gist.github.com/brabeji/454c70f8612a2bdf0b41e79cdd14658d)

**Comment by gpbl** [comment]
Closing this, as the issue has been fixed in v0.0.9 some time ago :)
